src/ds5/ds5-options.cpp

```diff
@@ -732,12 +734,62 @@ namespace librealsense
         if (res.empty())
             throw invalid_value_exception("auto_exposure_limit_option::query result is empty!");
 
-        return *(reinterpret_cast<uint32_t*>(res.data() + 4));
+        return static_cast<float>(*(reinterpret_cast<uint32_t*>(res.data() + 4)));
     }
 
     option_range auto_gain_limit_option::get_range() const
     {
         return *_range;
     }
 
+    librealsense::thermal_compensation::thermal_compensation(
+        std::shared_ptr<ds5_thermal_monitor> monitor,
+        std::shared_ptr<option> toggle) :
+        _thermal_monitor(monitor),
+        _thermal_toggle(toggle)
+    {
+        try {
+            auto on = static_cast<bool>(_thermal_toggle->query());
+            _thermal_monitor->update(on);
+        }
+        catch (...)
+        {
+            LOG_WARNING("Initial Thermal Compensation control state could not be established");
+        }
+    }
+
+    float librealsense::thermal_compensation::query(void) const
+    {
+        auto val = _thermal_toggle->query();
+        _thermal_monitor->update(static_cast<bool>(val));
```

Comment:
I don't see why a const function that queries for a value would do an update...

Reply:
the member is defined as mutable

Reply:
But why? I see no need. Now you have to explain the mutable, and it does not make sense.

Reply:
Adjusted